tests/integration/common/s2n_test_common.py

```diff
@@ -122,7 +122,7 @@ def __test(scenario):
                 result = Result("Server process crashed")
 
         except AssertionError as error:
-            result = Result(error)
+            result = Result(str(error))
         finally:
```

Comment:
looking at this:

What happens is an exception other than `AssertionError` happens? should we also match for `except BaseException as error:` here?

Reply:
I think we don’t want a catch all here since the AssertionErrors thrown here are only thrown very specifically for unexpected stdout from openssl and s2n. If we were to catch everything here, we would also catch runtime errors that occur due to a test that’s written incorrectly, like type errors or dividing by 0 or something, where we would probably want a runtime error to happen rather than printing it out and continuing. Unless we do actually want to print it out and continue no matter what?

If we want to catch more than just AssertionErrors, we would maybe want to create a custom exception that we catch, and then subclass it with our own AssertionError equivalent. And then the tests could also throw their own exceptions that get printed but continue to the next test. But since none of the tests raise their own exceptions maybe it’s fine to just catch AssertionErrors.